src/server/scripts/DragonIsles/AberrusTheShadowedCrucible/boss_kazzara_the_hellforged.cpp

```diff
@@ -93,10 +274,448 @@ struct boss_kazzara_the_hellforged : public BossAI
+        events.ScheduleEvent(EVENT_RAYS_OF_ANGUISH, 100ms, 400ms);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
+            return;
+
+        events.Update(diff);
+
+        while (uint32 eventId = events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_MOLTEN_SCAR:
+                    DoCastSelf(SPELL_MOLTEN_SCAR_AREATRIGGER);
+                    events.ScheduleEvent(EVENT_RAYS_OF_ANGUISH, 100ms, 400ms);
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+private:
+    EventMap events;
+};
+
+// 406525 - Dread Rift
+class spell_kazzara_dread_rift : public AuraScript
+{
+    void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        GetTarget()->CastSpell(GetTarget()->GetPosition(), SPELL_DREAD_RIFT_SUMMON, true);
+    }
+
+    void Register() override
+    {
+        AfterEffectRemove += AuraEffectRemoveFn(spell_kazzara_dread_rift::OnRemove, EFFECT_0, SPELL_AURA_MOD_SCALE, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+// 406517 - Dread Rift (periodic selector)
+class spell_kazzara_dread_rift_player_select : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo(
+        {
+            SPELL_DREAD_RIFT_SELECTOR,
+            SPELL_DREAD_RIFT_AURA
+        });
+    }
+
+    void FilterTargets(std::list<WorldObject*>& targets)
+    {
+        // @TODO: remove tank unless alive players < x
+        targets.remove_if([](WorldObject* target) -> bool
+        {
+            Player* player = target->ToPlayer();
+
+            if (!player)
+                return true;
+
+            if (player->HasAura(SPELL_DREAD_RIFT_SELECTOR))
+                return true;
+
+            return false;
+        });
+    }
+
+    void HandleDreadRift(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetHitUnit(), SPELL_DREAD_RIFT_AURA, true);
+    }
+
+    void HandleDreadRiftSelector(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetHitUnit(), SPELL_DREAD_RIFT_SELECTOR, true);
+    }
+
+    void Register() override
+    {
+        OnObjectAreaTargetSelect += SpellObjectAreaTargetSelectFn(spell_kazzara_dread_rift_player_select::FilterTargets, EFFECT_ALL, TARGET_UNIT_SRC_AREA_ENEMY);
+        OnEffectHitTarget += SpellEffectFn(spell_kazzara_dread_rift_player_select::HandleDreadRift, EFFECT_0, SPELL_EFFECT_DUMMY);
+        OnEffectHitTarget += SpellEffectFn(spell_kazzara_dread_rift_player_select::HandleDreadRiftSelector, EFFECT_1, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 407196 - Dread Rifts
+class spell_kazzara_dread_rifts : public SpellScript
+{
+    void HandleHit(SpellEffIndex /*effIndex*/)
+    {
+        uint32 spellId = SPELL_DREAD_RIFTS_3_TARGETS;
+
+        Unit::AuraApplicationMap const& appliedAuras = GetCaster()->GetAppliedAuras();
```

Comment:
Since you're removing these auras when applying the next one upon reaching a certain healthPct, you can just check if HasAura. There's no need to make a map for appliedAuras.